### PR TITLE
Fix https timeout

### PR DESCRIPTION
**Pull request by Shine-neko**
fix https://github.com/sozu-proxy/sozu/issues/787

### CODE REVIEW

**Review by Keksoj — CHANGES_REQUESTED** [review]
this is good. can you please change the commit message to be a bit more explicit? for instance, *start_https_worker: send empty listeners* or similarly

File: lib/src/https.rs [hunk 1]
```diff
@@ -1567,6 +1569,14 @@ pub fn start_https_worker(
     {
         let (scm_server, _scm_client) =
             UnixStream::pair().with_context(|| "Failed at creating scm stream sockets")?;
+        let scm = ScmSocket::new(_scm_client.into_raw_fd());
+        if let Err(e) = scm?.send_listeners(&Listeners {
```
[the remark this answers is not in the document]
Comment (Shine-neko) [reply lib/src/https.rs]: @Keksoj Listeners has no `default()`.  Should i add a `#[derive(Default)]` ?

[the remark this answers is not in the document]
Comment (Keksoj) [reply lib/src/https.rs]: what about this line? `Listeners::default()` works in other places of the code and your commit compiles

https://github.com/sozu-proxy/sozu/blob/66114a495625d4f6d3c2586d39d27b74dc9b426e/command/src/scm_socket.rs#L212-L213

[the remark this answers is not in the document]
Comment (Shine-neko) [reply lib/src/https.rs]: this example does not compile https://github.com/Shine-neko/proxie  :thinking:

[the remark this answers is not in the document]
Comment (Keksoj) [reply lib/src/https.rs]: the implementation of Default is very recent, your branch on your fork has not fetched it. But it works in the CI ;-)

**Comment by Keksoj** [comment]
you have renamed the commit `start_https_worker: send empty listeners or similarly` please remove `or similarly` @Shine-neko

File: lib/src/https.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Shine-neko) [reply lib/src/https.rs]: Ha okay perfect :)

**Comment by FlorentinDUBOIS** [comment]
@Wonshtrum could you take a look at it?

**Comment by Wonshtrum** [comment]
Hi @Shine-neko thanks for contributing to Sozu and pointing out the missing `send_listeners` (which blocks the server start). I'm sorry we closed your original issue without proper investigation.
I revisited this old code we never use and found that the TCP proxy had the same problem. To avoid future problems I merged the bulk of the 3 `start_*_worker` functions in `prebuild_server` in #1068. I added you as a co-author I hope this is fine.

**Comment by Keksoj** [comment]
Closing this for the issue is fixed by #1068 
Thanks for raising our awareness on the issue and your contributing, @Shine-neko !

**Pull request closed without merging**